doc/reference/reactivity.md

```diff
@@ -64,68 +107,237 @@ The function will be called whenever any tracked value is updated.
 
-Given a reactive object, this function returns the underlying, non-reactive,
-corresponding object.
+### `toRaw`
+
+While `markRaw` marks an object so that it is never made reactive, `toRaw` takes an object and returns
+the underlying non-reactive object. It can be useful in some niche cases. In particular, because the
+reactivity system returns a proxy, the returned object does not compare equal to the original object:
 
 ```js
-// in setup
-const state = useState({ value: 1 });
+const obj = {};
+const reactiveObj = reactive(obj);
+console.log(obj === reactiveObj); // false
+console.log(toRaw(obj) === reactiveObj); // true
+```
+
+It can also be useful during debugging, as unfolding proxies recursively in debuggers can be confusing.
+
+## Advanced usage
+
+The following is a collection of small snippets that leverage the reactivity system in "non-standard"
+ways to help you understand its power and where using it might make your code simpler.
+
+### Notification manager
 
-// later:
-const rawState = toRaw(this.state);
-rawState.value = 3; // will NOT be picked up by the reactivity system!!!
+Showing notifications is a pretty common need in web applications, you may want to show a notification
+from any other component within the application, and the notifications should stack on top of one another
+regardless of which component spawned them, here is how we can leverage the reactivity to accomplish this:
+
+```js
+let notificationId = 1;
+const notifications = reactive({});
+class NotificationContainer extends Component {
+  static template = xml`
+    <t t-foreach="notifications" t-as="notification" t-key="notification_key" t-esc="notification"/>
+  `;
+  setup() {
+    this.notifications = useState(notifications);
+  }
+}
+
+export function addNotification(label) {
+  const id = notificationId++;
+  notifications[id] = label;
+  return () => {
+    delete notifications[id];
+  };
+}
 ```
 
-Here again, this is useful in some situations where we want to explicitely bypass
-Owl, but using this function means that the responsability of coordinating
-state update is given to the user code, instead of Owl. Subtle bugs may arise!
+Here, the `notifications` variable is a reactive object. Notice how we didn't give `reactive` a callback:
+this is because in this case, all we care about is that adding or removing notifications in the
+`addNotification` function goes through the reactivity system. The `NotificationContainer` component
+reobserves this object with `useState`, and will be updated whenever notifications are added or removed.
+
+### Store
 
-Also, normal (non-reactive objects) will be directly returned by `toRaw`:
+Centralizing application state is a pretty common want/need in web applications. Because of the way
+the reactivity system works, you can treat any reactive object as a store, and if you call `useState`
+on it, components will automatically observe only the part of the store that they're interested in:
```

Comment:
nice example. also notice that it works with any class, and for now, I think that it's a good pattern:
```js
class TaskList {
  constructor() {
    ...
  }

 addTask() {...}
  toggleTask(){ ...}
}

// later, in a service in odoo
start() {
  const store = reactive(new TaskList());
  return store;
}

// and a utility hook:
function useTaskList() {
  return useState(useService("task"));
}

// and voila, components have a reactive task list that works as a store, but they can call methods on it
```
and